src/NuGet.Clients/NuGet.Console/PowerConsoleToolWindow.cs

```diff
@@ -146,20 +145,22 @@ public override void OnToolWindowCreated()
         {
             base.OnToolWindowCreated();
 
-            // Register key bindings to use in the editor
-            var windowFrame = (IVsWindowFrame)Frame;
-            if (windowFrame != null)
-            {
-                // Set F1 help keyword
-                WindowFrameHelper.AddF1HelpKeyword(windowFrame, keywordValue: F1KeywordValuePmc);
-            }
-            var cmdUi = VSConstants.GUID_TextEditorFactory;
-            windowFrame.SetGuidProperty((int)__VSFPROPID.VSFPROPID_InheritKeyBindings, ref cmdUi);
-
             // start a task when VS us idle and dont await it immediately
-            _loadTask = ThreadHelper.JoinableTaskFactory.StartOnIdle(
+            _loadTask = NuGetUIThreadHelper.JoinableTaskFactory.StartOnIdle(
                 async () =>
                 {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                    // Register key bindings to use in the editor
```

Comment:
StartOnIdle always invokes the delegate to the main thread, so this is not necessary. I suppose you added this to eliminate VSTHRD010 warnings though. I filed [https://github.com/Microsoft/vs-threading/issues/286](https://github.com/Microsoft/vs-threading/issues/286) to track improving this so you don't have to do this.

Reply:
Should i just keep the warning Suppress message then? or wouldn't this be a quick noop when it's already on UI thread? so may be keep the same code?

Reply:
I would keep what you have here. It's a fast no-op, and a bit more elegant (IMO) than a `#pragma warning disable` line pair in your code.